server/src/main/java/org/elasticsearch/cluster/routing/allocation/allocator/BalancedShardsAllocator.java

```diff
@@ -141,6 +143,35 @@ public ShardAllocationDecision decideShardAllocation(final ShardRouting shard, f
         return new ShardAllocationDecision(allocateUnassignedDecision, moveDecision);
     }
 
+    /**
+     * This method is called when there are no data nodes in the cluster.
+     *
+     * Newly created unassigned primary shards, with no prior allocation attempts
+     * are classified as yellow instead of red by cluster health.
+     * This function explicitly sets their allocation status to DECIDERS_NO, to
+     * indicate red indices with unassigned shards.
+     */
+    private void setAllocationStatus(RoutingAllocation allocation){
+        RoutingNodes routingNodes = allocation.routingNodes();
+        RoutingNodes.UnassignedShards unassignedShards = routingNodes.unassigned();
+        if (unassignedShards.isEmpty()) {
+            return;
+        }
+        RoutingNodes.UnassignedShards.UnassignedIterator unassignedIterator = unassignedShards.iterator();
+        while (unassignedIterator.hasNext()) {
+            ShardRouting shard = unassignedIterator.next();
+            UnassignedInfo shardInfo = shard.unassignedInfo();
```

Comment:
`shardInfo` is a bit of an ambiguous name for a variable.

```suggestion
            UnassignedInfo unassignedInfo = shard.unassignedInfo();
```